mayastor/tests/error_count_retry.rs

```diff
@@ -1,139 +1,141 @@
+async fn nexus_retry_child_write_succeed_test() {
+    let nexus_name = "error_retry_write_succeed";
+    let error_device = "error_device_write_succeed";
+    let ee_error_device = format!("EE_{}", error_device);
+    let bdev_ee_error_device = format!("bdev:///{}", ee_error_device);
+
+    common::truncate_file(ERROR_DISK, 64 * 1024);
+    common::truncate_file(NON_ERROR_DISK, 64 * 1024);
+
+    mayastor()
+        .spawn(async move {
+            create_error_bdev(error_device, ERROR_DISK);
+            create_nexus(
+                nexus_name,
+                &bdev_ee_error_device,
+                &NON_ERROR_BASE_BDEV,
+            )
+            .await;
+
+            check_nexus_state_is(nexus_name, NexusStatus::Online);
+
+            inject_error(
+                &ee_error_device,
+                SPDK_BDEV_IO_TYPE_WRITE,
+                VBDEV_IO_FAILURE,
+                1,
+            );
+
+            err_write_nexus(nexus_name, true).await; //should succeed, 2 attempts vs 1 error
+            check_nexus_state_is(nexus_name, NexusStatus::Degraded);
+            delete_nexus(nexus_name).await;
+        })
+        .await;
+
+    common::delete_file(&[ERROR_DISK.to_string()]);
+    common::delete_file(&[NON_ERROR_DISK.to_string()]);
     common::delete_file(&[YAML_CONFIG_FILE.to_string()]);
 }
 
-async fn create_nexus() {
-    let ch = vec![BDEV_EE_ERROR_DEVICE.to_string()];
+#[tokio::test]
+async fn nexus_retry_child_write_fail_test() {
+    let nexus_name = "error_retry_write_fail";
+    let error_device = "error_device_write_fail";
+    let ee_error_device = format!("EE_{}", error_device);
+    let bdev_ee_error_device = format!("bdev:///{}", ee_error_device);
+
+    common::truncate_file(ERROR_DISK, 64 * 1024);
+    common::truncate_file(NON_ERROR_DISK, 64 * 1024);
+
+    mayastor()
+        .spawn(async move {
+            create_error_bdev(error_device, ERROR_DISK);
+            create_nexus(
+                nexus_name,
+                &bdev_ee_error_device,
+                &NON_ERROR_BASE_BDEV,
+            )
+            .await;
+            check_nexus_state_is(nexus_name, NexusStatus::Online);
+
+            inject_error(
+                &ee_error_device,
+                SPDK_BDEV_IO_TYPE_WRITE,
+                VBDEV_IO_FAILURE,
+                2,
+            );
+
+            err_write_nexus(nexus_name, false).await; //should fail, 2 attempts vs 2 errors
+            check_nexus_state_is(nexus_name, NexusStatus::Degraded);
+            delete_nexus(nexus_name).await;
+        })
+        .await;
+
+    common::delete_file(&[ERROR_DISK.to_string()]);
+    common::delete_file(&[NON_ERROR_DISK.to_string()]);
+    common::delete_file(&[YAML_CONFIG_FILE.to_string()]);
+}
+
+fn check_nexus_state_is(name: &str, expected_status: NexusStatus) {
+    let nexus = nexus_lookup(name).unwrap();
+    assert_eq!(nexus.status(), expected_status);
+}
+
+async fn create_nexus(name: &str, err_dev: &str, dev: &str) {
```

Comment:
You could make this function receive a vector, then you could pass in any number of children.